cupy/_core/core.pyx

```diff
@@ -1895,6 +1895,10 @@ cdef list _cupy_extra_header_list = [
     'cupy/complex/csqrtf.h',
     'cupy/complex/catrig.h',
     'cupy/complex/catrigf.h',
+    'cupy/swap.cuh',
+    'cupy/tuple/type_traits.h',
+    'cupy/tuple/tuple.h',
+    'cupy/tuple.cuh',
```

Comment:
For HIP, because hipRTC does not support `-I` to search headers (https://github.com/ROCm-Developer-Tools/HIP/issues/2182), we need to provide a list of all headers that are included. Currently this is done unconditionally, but we could improve it in the future. (Or better: ask AMD to fix it...)